src/KustoExpressionParser.ts

```diff
@@ -43,17 +43,25 @@ export class KustoExpressionParser {
     };
 
     const parts: string[] = [];
+    const isDynamic = query.search.property.name.includes(DYNAMIC_TYPE_ARRAY_DELIMITER);
+    const name: string = this.addExpanPartsdIfNeeded(query.search.property.name, parts);
     this.appendProperty(context, query.expression.from, parts);
     this.appendTimeFilter(context, undefined, parts, tableSchema);
 
-    if (query.index) {
+    //query.index is used by the legacy query editor
+    if (query.index && !isDynamic) {
```

Comment:
I just noticed the current code has a bug, if there is no index, we are not appending the condition `isnotempty(col)`. If query.index is not defined, it should still append that.

Reply:
This get's appended [here](https://github.com/grafana/azure-data-explorer-datasource/blob/df669efb46ba75232491ae829ffe0ac0249d7b5e/src/KustoExpressionParser.ts#L283) if I'm understanding this correctly. Should already be good to go.